src/legacy/core_plugins/metrics/public/np/index.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Plugin } from './plugin';
+
+new Plugin().setup();
```

Comment:
> [..] the convention for the index.ts and plugin.ts files are that they are in the top-level of the `/public` and `/server` directories.

That is true for NP plugins, but his is a shim. I don't remember what were the folder guidelines for shims, if any.

> If this was placed in a separate folder to avoid confusion...

In a legacy plugin, which has old code as well as a shim, I would probably also put it into a separate folder, as to not mix legacy platform code with NP code. For example, APM plugin does something similar, they just call it `new-platform` instead of `np`:

- `apm/public/new-platform/index.ts`
- `apm/public/new-platform/plugin.ts`
- `apm/server/new-platform/index.ts`
- `apm/server/new-platform/plugin.ts`

Reply:
i think we want to have the same structure as for NP plugins, so we can easily move them over later.

what do you mean with a plugin that has `old code as well as shim`?
tsvb does 3 things .... it registers visType and associated interpreter function. 
And it provides server endpoint.  <-- this is the part that we wouldn't be moving yet.
so everything in public can be updated.

i agree with moving this one level up. i would also move everything in kbn_vis_type one level up.

Reply:
Folder structure has been changed.

Reply:
@ppisljar 

> what do you mean with a plugin that has `old code as well as shim`?

I mean if a legacy plugin is in progress of shimming, there will be legacy plugin code as well as shimmed plugin code. I would personally prefer having the shim in a separate folder, where I know that that folder is a shim and it can basically be copy-pasted to the NP.

Though, if this plugin is simple that it can all be shimmed in one go, then there is no legacy platform code, so having an extra folder is not useful.

Reply:
The APM directories contain more than just "shims" -- the shims are actually mixed with NP code. We've done the same thing in the `data` plugin, which in hindsight is just confusing, and something I'm working on a PR to clean up so we have a clearer separation of legacy/shim code, and NP code.

The idea with the shims is that you could have a `plugin.ts` file that's "pure" new platform code -- written exactly as it will be in the new platform -- and you can have a separate shim file that does all of the legacy imports and calls `new Plugin.setup(...shimmedDependencies)`, mimicking what NP will do for you.

That shim file is what would be loaded by `hacks` (or imported temporarily by other downstream plugins, likely inside of __their__ shims), and can simply be deleted at the end of the migration process and the rest of the code should "just work" in the NP.

In most cases, the shim logic you need can probably be contained in a single file, but I totally agree it's appropriate to have a separate folder if there's a more complex shim. My only goal is that at the end of migration, something labeled `shim` means `'safe to delete'`